kymatio/scattering2d/frontend/torch_frontend.py

```diff
@@ -0,0 +1,110 @@
+__all__ = ['Scattering2DTorch']
+
+import torch
+import torch.nn as nn
+
+from .base_frontend import Scattering2DBase
+from kymatio.scattering2d.core.scattering2d import scattering2d
+from ...frontend.torch_frontend import ScatteringTorch
+
+
+class Scattering2DTorch(ScatteringTorch, Scattering2DBase):
+    def __init__(self, J, shape, L=8, max_order=2, pre_pad=False,
+    backend='torch'):
+        ScatteringTorch.__init__(self)
+        Scattering2DBase.__init__(self, J, shape, L, max_order, pre_pad, backend)
+        Scattering2DBase._instantiate_backend(self,
+                'kymatio.scattering2d.backend.')
+        Scattering2DBase.build(self)
+        Scattering2DBase.create_filters(self)
+
+        self.register_filters()
+
+
+    def backend_setup(self):
+        if not self.backend:
+            from ..backend.torch_backend import backend
+            self.backend = backend
+        elif not self.backend.name.startswith('torch'):
+            raise RuntimeError('This backend is not supported.')
+        
+
+    def register_filters(self):
+        """ This function run the filterbank function that
+            will create the filters as numpy array, and then, it
+            saves those arrays as module's buffers."""
+
+        # Create the filters
+        n = 0
+        for c, phi in self.phi.items():
+            if isinstance(c, int):
+                self.phi[c] = torch.from_numpy(self.phi[c]).unsqueeze(-1) # add a trailing singleton dimension to mark
+                # it as non-complex
+                self.register_buffer('tensor' + str(n), self.phi[c])
+                n += 1
+
+        for j in range(len(self.psi)):
+            for k, v in self.psi[j].items():
+                if isinstance(k, int):
+                    self.psi[j][k] = torch.from_numpy(v).unsqueeze(-1) # add a trailing singleton dimension to mark it
+                    # as non-complex
+                    self.register_buffer('tensor' + str(n), self.psi[j][k])
+                    n += 1
+
+    def scattering(self, input):
+        # each time scattering is run, one needs to make sure self.psi and self.phi point to
+        # the correct buffers
+        n = 0
+        buffer_dict = dict(self.named_buffers())
+        for c, phi in self.phi.items():
+            if isinstance(c, int):
+                self.phi[c] =  buffer_dict['tensor' + str(n)]
+                n += 1
+
+        for j in range(len(self.psi)):
+            for k, v in self.psi[j].items():
+                if isinstance(k, int):
+                    self.psi[j][k] = buffer_dict['tensor' + str(n)]
+                    n += 1
```

Comment:
Possibly put into own function? Have scattering be a wrapper function?

Reply:
I think it's ok to make it appearant, as we should target to remove this later!